packages/client-python/src/rocketride/client.py

```
@@ -164,28 +164,35 @@ def __init__(
-                                self._env.setdefault(key, value)
+                                self._env[key] = value
+                                config_env[key] = value
             except Exception:
                 # File doesn't exist or can't be read - that's okay
                 pass
+
+            # Connection bootstrap should honor non-empty process env values without
+            # changing the older substitution behavior that reads from self._env.
+            for key, value in os.environ.items():
+                if value:
+                    config_env[key] = value
```

Ruff (0.15.7)

[warning] Use a dictionary comprehension instead of a for-loop

(PERF403)